### PR TITLE
Tweak proto file

### PR DESCRIPTION
**Pull request by pseudomuto**
There are a few things about the auto-generated proto file that I wasn't a big fan of. Particularly:

* There is a shared `Reference` object used to encapsulate an id field. While this save some duplication, it means that we can't alter any of those endpoints in the future without affecting all of them (e.g. if one of them took an extra field or something).
* Some of the endpoints use `google.protobuf.Empty`. If we wanted to add something like pagination in the future, we'd break all existing clients (since we'd change the request message type).

**What I've Done Here**

* Added version to the package name and set `go_package` option
* Ensured that each RPC has it's own request/response message (allows for easy evolution without typing them together)
* Removed all the built-in protobuf protos (they're already available in lib path if you've got protoc installed)

### CODE REVIEW

File: proto/swapi.proto [hunk 1]
```diff
@@ -1,9 +1,83 @@
...
+service Starwars {
+  rpc GetFilm(GetFilmRequest) returns (GetFilmResponse) {
+    option (google.api.http) = {
+      get: "/swapi/v1/film/{id}"
+    };
+  }
+
+  rpc ListFilms(ListFilmsRequest) returns (ListFilmsResponse) {
+    option (google.api.http) = {
+      get: "/swapi/v1/film/{id}"
```
Comment (jhump) [inline proto/swapi.proto]: This is the same as for `GetFilm`. I assume this was supposed to be `/swapi/v1/films`?

File: proto/swapi.proto [hunk 2]
```diff
@@ -1,9 +1,83 @@
...
+
+  rpc ListPlanets(ListPlanetsRequest) returns (ListPlanetsResponse) {
+    option (google.api.http) = {
+       get: "/swapi/v1/planets"
+    };
+  }
+
+  rpc GetPerson(GetPersonRequest) returns (GetPersonResponse) {
+    option (google.api.http) = {
+       get: "/swapi/v1/person/{id}"
```
Comment (jhump) [inline proto/swapi.proto]: Is it common to switch URI path elements form plural to singular fo this? In my experience, the URI still indicates plural -- e.g. the directory containing people, and that the ID then uniquely identifies a single person. (Here and throughout.)

Or does the existing SWAPI do this, and you're just mimic'ing that?

**Review by jhump — COMMENTED** [review]
LGTM

**Review by konsumer — COMMENTED** [review]
I added the proto/google folder to better facilitate tools that can only use a single include dir, and make includes & dependencies more explicit. I am open to removing anything we don't use, but also I kinda like to have the folder around to give myself a toolbox of basic well-known types and useful annotations (like google.http).

I think we should be using WKT Timestamp for the dates that are around. There are probably a few others that really should be WKT.

Why are the grpc endpoints being removed? They've been implemented in node. They have been helpful to me in troubleshooting the data-interconnections and cross-validating type connections.

File: proto/swapi.proto (hunk 2 above)
Comment (konsumer) [reply proto/swapi.proto]: This was my code in another commit, but I can answer your question: swapi uses a plural noun for all. I am fine with changing it to that.

File: proto/swapi.proto (hunk 1 above)
Comment (konsumer) [reply proto/swapi.proto]: Not sure I understand your comment.

**Comment by pseudomuto** [comment]
> I think we should be using WKT Timestamp for the dates that are around

I'm not familiar with WKT Timestamps...anything special? I was thinking of switching these to `google.protobuf.Timestamp` objects.

> Why are the grpc endpoints being removed?

Which ones? I certainly didn't think I was removing endpoints 😮

**Comment by konsumer** [comment]
`google.protobuf.Timestamp` is a WKT ([well known type](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf).) it's a collection of common types that are often used. I dunno if they are "special", just kind of standard. And they don't perfectly fit with my preferences, for example, I'd prefer time to be in ms (not nanos) and some text-readable format (ISO 8601 is nice because it's string-sortable and very common) but I use them because it's a standard, and they even have suggested JSON mapping counterparts, which can make following the standards more friendly.

I think I misread the diff. I was evaluating the PR before I had my coffee. :) you didn't remove any endpoints, sorry I misunderstood.

**Comment by konsumer** [comment]
LGTM, and I can make an issue for "switch time representations to Timestamp" and if no one objects, add the [well known types](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf) back.  It's a bit of a pain to have to include them in the same proto dir, but I have gotten used to that limitation of the node grpc library (and protobuf library) and it can be kinda nice to have all the definitions in one place, ready for use, and super-explicit.

File: proto/swapi.proto (hunk 2 above)
Comment (pseudomuto) [reply proto/swapi.proto]: I'll update everything to plural 👍

**Comment by pseudomuto** [comment]
Updated to ensure all http endpoints use plural nouns. I've also added the WKT protos back in

**Comment by jhump** [comment]
@konsumer, what tools exactly only allow a single import path? It seems like an anti-pattern to me to include a copy of them in every project. Instead, you should be able to pin the version of protoc that the project uses, and protoc then includes its own copy of these files.

(The only exception is for the HTTP gateway annotations, which are not included with protoc.)

**Comment by pseudomuto** [comment]
> Instead, you should be able to pin the version of protoc that the project uses, and protoc then includes its own copy of these files

FWIW, that's been my take on it as well

**Comment by pseudomuto** [comment]
Is this good to go? We can still continue the keep/remove WKT types in an issue if necessary

**Review by jhump — APPROVED** [review]
> Is this good to go?

I think it looks good.

**Comment by konsumer** [comment]
> Is this good to go?

Yep, I think so, too.

@jhump all the dynamic javascript tools only allow a single include-root, including protobufjs which is what the official grpc lib uses, in node. It feels like an anti-pattern to me too, but it's not unlike Go's way of having a single place for includes/libraries in a shape that you can expect. Only some of the google stuff come with protoc (for example, the http annotations are missing, as you noted,) so it's not a given all the basics we are using would be installed. 

In local code, I am mostly doing it all dynamic, with no generated interfaces. An alternate method to including them in the repo would be to use git submodule, but I find it's just easier to just include the protos in the parent project I'm working on to get the IDLs working on new dev's machines as fast as possible. Maybe it ties back to how I generally use grpc, which is a single org-wide model source-of-truth, and keeping all the referenced proto files together makes everything much simpler and end-to-end self-referential (like even the proto describing proto is included.)

**Pull request merged**